qemu/tests/virtio_console.py

```diff
@@ -37,6 +39,81 @@ def __set_exit_event():
     EXIT_EVENT.set()
 
 
+def add_chardev(vm, params):
+    """
+    Generate extra CharDevice without serial device utilize it
+
+    :param vm: VM object to be operated
+    :param params: Dictionary with the test parameters
+    :return list of added CharDevice object
+    """
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    qdevices = qcontainer.DevContainer(qemu_binary, vm.name,
+                                       params.get('strict_mode'),
+                                       params.get(
+                                           'workaround_qemu_qmp_crash'),
+                                       params.get('allow_hotplugged_vm'))
+    char_devices = params['extra_chardevs'].split()
+    host = params.get('chardev_host', '127.0.0.1')
+    free_ports = utils_misc.find_free_ports(
+        5000, 6000, len(char_devices), host)
+    device_list = []
+    for index, chardev in enumerate(char_devices):
+        chardev_param = params.object_params(chardev)
+        file_name = vm.get_serial_console_filename(chardev)
+        backend = chardev_param.get('chardev_backend',
+                                    'unix_socket')
+        if backend in ['udp', 'tcp_socket']:
+            chardev_param['chardev_host'] = host
+            chardev_param['chardev_port'] = free_ports[index]
```

Comment:
‘chardev_port’ type needs to be a string while doing hotplug.

07:23:04 ERROR|   File "/home/kar/kar/workspace/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/virtio_serial_hotplug_port_various_chardev.py", line 39, in run
07:23:04 ERROR|     vm.devices.simple_hotplug(chardevice, vm.monitor)
07:23:04 ERROR|   File "/home/kar/kar/workspace/avocado-vt/virttest/qemu_devices/qcontainer.py", line 678, in simple_hotplug
07:23:04 ERROR|     out = device.hotplug(monitor)
07:23:04 ERROR|   File "/home/kar/kar/workspace/avocado-vt/virttest/qemu_devices/qdevices.py", line 257, in hotplug
07:23:04 ERROR|     return monitor.cmd(cmd, args)
07:23:04 ERROR|   File "/home/kar/kar/workspace/avocado-vt/virttest/qemu_monitor.py", line 1943, in cmd
07:23:04 ERROR|     raise QMPCmdError(cmd, args, r["error"])
07:23:04 ERROR| virttest.qemu_monitor.QMPCmdError: QMP command 'chardev-add' failed    (arguments: {'id': 'channel2', 'backend': {'type': 'socket', 'data': {'addr': {'type': 'inet', 'data': {'host': '127.0.0.1', 'port': 5142}}}}},    error message: {'class': 'GenericError', 'desc': "Invalid parameter type for 'backend.data.addr.data.port', expected: string"}